.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x]
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - uses: pnpm/action-setup@v4
+      with:
+        version: 9
+    
```

Comment:
create an action just like https://github.com/rip-technologies/shipstation-node/commit/101b4e29f8b7a7344e1447adcbfd69a253f238e8 and ensure that pnpm is desclared inside the root package.json. 

also add this to the package.json:
```
"packageManager": "pnpm@10.11.0"
```

Reply:
this should also have the Install Depedencies step so that it uses ` wyvox/action-setup-pnpm`